lib/chef/knife/core/status_presenter.rb

```diff
@@ -100,7 +100,14 @@ def summarize(list)
             fqdn = (node[:ec2] && node[:ec2][:public_hostname]) || node[:fqdn]
             name = node["name"] || node.name
 
-            run_list = (node["run_list"]).to_s if config[:run_list]
+            if config[:run_list]
+              run_list = node["run_list"]
+              if config[:long_output]
+                run_list = Array.new
+                node.run_list.each { |rl| run_list << "#{rl.type}[#{rl.name}]" }
```

Comment:
It seems like map would make this whole bit a lot easier to read

Reply:
Added map instead of each